compiler/luci/pass/src/RemoveRedundantReshapePass.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveRedundantReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+bool remove_redundant_reshape(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleReshape *>(node);
+  if (target_node == nullptr)
+    return false;
+  auto pred_node = dynamic_cast<luci::CircleReshape *>(target_node->tensor());
+  if (pred_node == nullptr)
+    return false;
+  auto value_node = loco::must_cast<luci::CircleNode *>(pred_node->tensor());
+  target_node->tensor(value_node);
```

Comment:
This seems to disconnect `pred_node` without any conditions. Can you please explain about how this is possible?

Reply:
It check subsequent `Reshape` Operation and check `pred_node` doesn't have control inputs. I think I forget to check `pred_node` has control inputs. I will check this and add implementation. sorry for missing.

Reply:
I find some information about `Reshape` in TFlite kernel. on this [operation](https://github.com/tensorflow/tensorflow/blob/fefcd5704ba4efe873485b981644ce0fb18c2440/tensorflow/lite/kernels/reshape.cc#L47-L50) number of input elements and number of output elements should be same. so, I think this check only reasonable on tensorflow grappler, I think.

Reply:
You mean, current code does not need additional condition checks? 
Can you add some comments about how `nodeA`-`reshapeB`-`reshapeC` can be just changed to `reshapeC` with an example?
For example, let's say output of `nodeA` is `(4,32)` and `reshapeB` make this `(2,2,32)` and then `reshapeC` makes this `(2,2,2,16)`. overall `(4,32)` willbe converted to `(2,2,2,16)` so `reshapeB` canbe removed.

Another question, with the new network, does this run in our runtime and tflite without any problems?

Reply:
> Can you add some comments about how `nodeA`-`reshapeB`-`reshapeC` can be just changed to `reshapeC` with an example?

A little bit different. fallow from reference, this will bypass `reshapeB` if another reshape(`reshapeC`) get input as `reshapeB` output.( `NodeA` goes to `ReshapeC` directly)
So, final result will be like below.

![image](https://user-images.githubusercontent.com/2858751/101431220-caa2e100-3949-11eb-8f3e-b47885196e6a.png)

After changing like that, ReshapeB would be removed if it have no output connection with other active node.

> For example, let's say output of `nodeA` is `(4,32)` and `reshapeB` make this `(2,2,32)` and then `reshapeC` makes this `(2,2,2,16)`. overall `(4,32)` will be converted to `(2,2,2,16)` so `reshapeB` can be removed.

Yes. but, `ReshapeB` will be remain if another node get this as `input Tensor`. 
Something Worried about is the case that reshape node's second input will be dynamic case...

I think only accepting this with both `ReshapeB` and `ReshapeC` have Second input with type `CircleConst`.

> Another question, with the new network, does this run in our runtime and tflite without any problems?

I will check about this more.
Sorry for bothering.